src/submitter.py

```diff
@@ -0,0 +1,399 @@
+         strings or a tuple of (filename, file-like objects with ``.read()``).
+
+    :returns: tuple of (bytes, headers dict)
+
+    """
+    # NOTE(willkg): This is the result of uuid.uuid4().hex. We just need a
+    # unique string to denote the boundary between parts in the payload.
+    boundary = '01659896d5dc42cabd7f3d8a3dcdd3bb'
+    output = io.BytesIO()
+
+    # Package up raw crash metadata--sort them so they're stable in the payload
+    for key, val in sorted(raw_crash.items()):
+        output.write(smart_bytes('--%s\r\n' % boundary))
+        output.write(
+            smart_bytes('Content-Disposition: form-data; name="%s"\r\n' % Header(key).encode())
+        )
+        output.write(b'Content-Type: text/plain; charset=utf-8\r\n')
+        output.write(b'\r\n')
+        output.write(smart_bytes(val))
+        output.write(b'\r\n')
+
+    # Insert dump data--sort them so they're stable in the payload
+    for name, data in sorted(dumps.items()):
+        output.write(smart_bytes('--%s\r\n' % boundary))
+
+        # dumps are sent as streams
+        output.write(
+            smart_bytes(
+                'Content-Disposition: form-data; name="%s"; filename="file.dump"\r\n' %
+                Header(name).encode()
+            )
+        )
+        output.write(b'Content-Type: application/octet-stream\r\n')
+        output.write(b'\r\n')
+        output.write(data)
+        output.write(b'\r\n')
+
+    # Add end boundary
+    output.write(('--%s--\r\n' % boundary).encode('utf-8'))
+    output = output.getvalue()
+
+    # Generate headers
+    headers = {
+        'Content-Type': 'multipart/form-data; boundary=%s' % boundary,
+        'Content-Length': str(len(output)),
+    }
+
+    return output, headers
+
+
+def handler(event, context):
+    accepted_records = []
+
+    logger.info('number of records: %d', len(event['Records']))
+    for record in event['Records']:
+        # Skip anything that's not an S3 ObjectCreated:put event
+        if record['eventSource'] != 'aws:s3' or record['eventName'] != 'ObjectCreated:Put':
+            continue
+
+        # Extract bucket name for debugging
+        bucket = record['s3']['bucket']['name']
+
+        # Extract crash id--if it's not a raw_crash object, skip it.
+        crash_id = extract_crash_id_from_record(record)
+        if crash_id is None:
+            continue
+
+        logger.debug('saw crash id: %s in %s', crash_id, bucket)
+
+        # Skip crashes that are marked DEFER
+        if get_antenna_throttle_result(crash_id) == DEFER:
+            statsd_incr('socorro.submitter.defer', value=1)
+            continue
+
+        # Throttle ACCEPTed records
+        if CONFIG.throttle < 100 and random.randint(0, 100) > CONFIG.throttle:
+            statsd_incr('socorro.submitter.throttled', value=1)
+            continue
+
+        accepted_records.append(crash_id)
+
+    # If we don't have anything to post, we're done!
+    if not accepted_records:
+        return
```

Comment:
Yeah... I kind of like doing this because I find the narrative clearer. Instead of it being implied that it does nothing if there are no accepted records, it's explicit. If I'm reading through this code, I hit this point and continue reading without having to worry about this scenario.

So, this is a style thing. If you feel strongly, I can take it out.